src/cpp/statistics/rtps/monitor-service/MonitorService.cpp

```diff
@@ -262,73 +262,77 @@ bool MonitorService::write_status(
                     }
                     case INCOMPATIBLE_QOS:
                     {
                         data.incompatible_qos_status(IncompatibleQoSStatus_s{});
-                        assert(status_queryable_.get_monitoring_status(local_entity_guid, data));
+                        status_retrieved = status_queryable_.get_monitoring_status(local_entity_guid, data);
                         break;
                     }
                     //Not triggered for the moment
                     case INCONSISTENT_TOPIC:
                     {
                         EPROSIMA_LOG_ERROR(MONITOR_SERVICE, "Inconsistent topic status not supported yet");
-                        assert(false);
+                        (void)local_entity_guid;
                         break;
                     }
                     case LIVELINESS_LOST:
                     {
                         data.liveliness_lost_status(LivelinessLostStatus_s{});
-                        assert(status_queryable_.get_monitoring_status(local_entity_guid, data));
+                        status_retrieved = status_queryable_.get_monitoring_status(local_entity_guid, data);
                         break;
                     }
                     case LIVELINESS_CHANGED:
                     {
                         data.liveliness_changed_status(LivelinessChangedStatus_s{});
-                        assert(status_queryable_.get_monitoring_status(local_entity_guid, data));
+                        status_retrieved = status_queryable_.get_monitoring_status(local_entity_guid, data);
                         break;
                     }
                     case DEADLINE_MISSED:
                     {
                         data.deadline_missed_status(DeadlineMissedStatus_s{});
-                        assert(status_queryable_.get_monitoring_status(local_entity_guid, data));
+                        status_retrieved = status_queryable_.get_monitoring_status(local_entity_guid, data);
                         break;
                     }
                     case SAMPLE_LOST:
                     {
                         data.sample_lost_status(SampleLostStatus_s{});
-                        assert(status_queryable_.get_monitoring_status(local_entity_guid, data));
+                        status_retrieved = status_queryable_.get_monitoring_status(local_entity_guid, data);
                         break;
                     }
                     default:
                     {
                         EPROSIMA_LOG_ERROR(MONITOR_SERVICE, "Referring to an unknown status");
-                        assert(false);
+                        (void)local_entity_guid;
                         break;
                     }
                 }
 
-                status_data.status_kind((StatusKind)i);
-                status_data.value(data);
-                add_change(status_data, false);
-
+                assert(status_retrieved);
+                if (status_retrieved)
```

Comment:
I suggest removing the condition if it is being asserted just above.

Reply:
If depending on the compilation the `assert` is removed, we should then avoid `add_change` if status could not be correctly retrieved. I think we should keep both. I will add a log in case is `false` and it could not be retrieved

Reply:
I have talked with Edu and he agrees with me that asserting and using a conditional checking that condition is incompatible. Either we are completely sure that a specific situation should never happen and therefore an assertion is done, or if there is some small possibility that a corner case occurs, then the condition should be kept.